src/test/regress/expected/failure_add_disable_node.out

```diff
@@ -0,0 +1,359 @@
+ (0,coordinator,"['Query(query=COPY public.user_table_XXXXXX TO STDOUT)']")
+(3 rows)
+
+-- verify node is not activated
+SELECT * FROM master_get_active_worker_nodes()
+ORDER BY 1, 2;
+ node_name | node_port 
+-----------+-----------
+ localhost |     57640
+(1 row)
+
+SELECT shardid, shardstate
+FROM pg_dist_placement p JOIN pg_dist_shard s USING (shardid)
+WHERE s.logicalrelid = 'user_table'::regclass
+ORDER BY placementid;
+ shardid | shardstate 
+---------+------------
+  200000 |          1
+(1 row)
+
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- master_remove_node fails when there are shards on that worker
+SELECT master_remove_node('localhost', :worker_1_port);
+ERROR:  you cannot remove the primary node of a node group which has shard placements
+-- drop event table and re-run remove
+DROP TABLE event_table;
+SELECT master_remove_node('localhost', :worker_1_port);
+ master_remove_node 
+--------------------
+ 
+(1 row)
+
+-- verify node is removed
+SELECT * FROM master_get_active_worker_nodes()
+ORDER BY 1, 2;
+ node_name | node_port 
+-----------+-----------
+ localhost |     57640
+(1 row)
+
+SELECT shardid, shardstate
+FROM pg_dist_placement p JOIN pg_dist_shard s USING (shardid)
+WHERE s.logicalrelid = 'user_table'::regclass
+ORDER BY placementid;
+ shardid | shardstate 
+---------+------------
+  200000 |          1
+(1 row)
+
+-- test master_add_inactive_node
+-- it does not create any network activity therefore can not
+-- be injected failure through network
+SELECT citus.clear_network_traffic();
+ clear_network_traffic 
+-----------------------
+ 
+(1 row)
+
+SELECT master_add_inactive_node('localhost', :worker_1_port);
+             master_add_inactive_node              
+---------------------------------------------------
+ (3,3,localhost,57637,default,f,f,primary,default)
+(1 row)
+
+SELECT master_remove_node('localhost', :worker_1_port);
+ master_remove_node 
+--------------------
+ 
+(1 row)
+
+-- verify there is no network traffic generated
+SELECT citus.dump_network_traffic();
+ dump_network_traffic 
+----------------------
+(0 rows)
+
+SELECT shardid, shardstate
+FROM pg_dist_placement p JOIN pg_dist_shard s USING (shardid)
+WHERE s.logicalrelid = 'user_table'::regclass
+ORDER BY placementid;
+ shardid | shardstate 
+---------+------------
+  200000 |          1
+(1 row)
+
+-- test master_add_node replicated a reference table
+-- to newly added node.
+SELECT citus.mitmproxy('conn.onQuery(query="COPY").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT citus.clear_network_traffic();
+ clear_network_traffic 
+-----------------------
+ 
+(1 row)
+
+SELECT master_add_node('localhost', :worker_1_port);
```

Comment:
I think adding/removing `worker_1_port` tests one side of things. The [proxy](https://github.com/citusdata/citus/pull/2119/files#diff-a7ce4fb174e1acce25b5bcd3121576a9R572) is on `:worker_2_port`, so we should be adding/removing that as well. In these tests, we're catching the `COPY .. STDOUT` command that `worker_2` executes due to `worker_append_table_to_shard` that is being executed on `worker_1`.

 Note that the coordinator sends bunch of other commands to the target node as well:
```SQL
Select master_add_node('localhost', 9702);
NOTICE:  Replicating reference table "ref" to the node localhost:9702
LOG:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 22, '2018-07-11 17:46:36.461851+03');
DETAIL:  on server localhost:9702
LOG:  issuing SELECT worker_apply_shard_ddl_command (102074, 'DROP TABLE IF EXISTS public.ref CASCADE')
DETAIL:  on server localhost:9702
LOG:  issuing SELECT worker_apply_shard_ddl_command (102074, 'CREATE TABLE public.ref (a integer)')
DETAIL:  on server localhost:9702
LOG:  issuing SELECT worker_append_table_to_shard ('public.ref_102074', 'public.ref_102074', 'localhost', 9701)
DETAIL:  on server localhost:9702
LOG:  issuing COMMIT
DETAIL:  on server localhost:9702
                 master_add_node                  
--------------------------------------------------
 (9,9,localhost,9702,default,f,t,primary,default)
```

And, we should probably run these tests in a seperate schema and add a test on failure during `CREATE SCHEMA IF NOT EXISTS` ?